### PR TITLE
Started working on new plugin remake

### PR DESCRIPTION
**Pull request by notangelmario**
Well, time to revive this project!

Fresh has finally implemented [the ability for plugins to create routes](https://deno.com/blog/fresh-1.3#adding-routes-andor-middlewares-from-plugins), so this makes
implementing `fresh-seo` look less like it's built with duct tape.

In my opinion we should keep the old functionality for backwards compatibility.

Other things to do:
- [x] Remove the import from the `deno.json` and put it in `deps.ts` (I put it there temporarily)
- [x] Implement glob matching in removing routes (see #22 )
- [x] Add route options to `include` option
- [x] Remove functionality to create setup in the old style

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { SitemapContext } from "./sitemap.ts";
+import { Manifest, Plugin } from "./types.ts";
+
+interface PluginOptions {
+	include: string[];
+	exclude: string[];
+}
```
Comment (sinyo-matu) [inline src/plugin.ts:7]: Could we make Options optional?
Users may not need to set them
```suggestion
interface PluginOptions {
	include?: string[];
	exclude?: string[];
}
```

File: src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+export const freshSEOPlugin = (manifest: Manifest, opts: PluginOptions): Plugin => {
```
Comment (sinyo-matu) [inline src/plugin.ts:9]:
```suggestion
export const freshSEOPlugin = (manifest: Manifest, opts: PluginOptions = {}): Plugin => {
```

**Review by sinyo-matu — COMMENTED** [review]
@notangelmario awesome work! 
It runs really smoothly with my Fresh website. 
And I think this really makes it effortless!

Here are my comments.
If you have any questions, just let me know.

**Review by sinyo-matu — COMMENTED** [review]
And I think we should delete The createSitemap related codes located https://github.com/xstevenyung/fresh-seo/blob/d3cc04b8eddbd930839033ace374f67ba5585231/src/initFile.ts#L86
because user have got the `/sitemap.xls` route provided by our plugin

**Comment by notangelmario** [comment]
Oh yeah, I should've created the pull request as a draft but it is what it is.

`include` and `exclude` should also be an array of routes and options of those routes.

And yeah, the init process should be removed, but it shouldn't interfere with old setups.

**Comment by notangelmario** [comment]
Alright I think it's pretty much done. Anything to add?

**Review by sinyo-matu — COMMENTED** [review]
@notangelmario 
A big Thank you sir!

**Review by sinyo-matu — APPROVED** [review]

**Comment by xstevenyung** [comment]
lfg! thx for the update, excited to finally have the plugin api implemented

**Pull request merged**